utils/S3UtilsMongoClient.js

```diff
@@ -196,6 +198,21 @@ class S3UtilsMongoClient extends MongoClientInterface {
                 errorString: err.toString(),
             });
             return callback(err);
+        } finally {
+            if (cursor && !cursor.closed) {
+                log.info('Finished processing cursor', {
+                    method: 'getObjectMDStats',
+                    bucketName,
+                });
+                cursor.close();
+            }
+            if (cursorUsersBucketCreationDates && !cursorUsersBucketCreationDates.closed) {
```

Comment:
this is not a cursor, but an array: so cannot be closed like this I guess?

would need to introduce an extra variable before converting to array... and ideally close the cursor ASAP once we have retrieved the array...

Reply:
Removed the line as per my last comment, cursor is closed right after the `toArray()` call
The code for the other cursor is kept, as indeed in case of error durring the processing, that may lead an open cursor.